crypto/keyring_streaming_test.go

```diff
@@ -0,0 +1,228 @@
+package crypto
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+const testFilename = "filename.txt"
+
+func TestKeyRing_EncryptStream(t *testing.T) {
+	keyRingPrivate, err := keyRingTestPrivate.Copy()
+	if err != nil {
+		t.Fatal("Expected no error while copying keyring, got:", err)
+	}
+	keyRingPublic, err := keyRingTestPublic.Copy()
+	if err != nil {
+		t.Fatal("Expected no error while copying keyring, got:", err)
+	}
+	messageBytes := []byte("Hello World!")
+	messageReader := bytes.NewReader(messageBytes)
+	var ciphertextBuf bytes.Buffer
+	isBinary := true
+	filename := "filename.txt"
+	modTime := GetUnixTime()
+	messageWriter, err := keyRingPublic.EncryptStream(
+		&ciphertextBuf,
+		isBinary,
+		filename,
+		modTime,
+		keyRingPrivate,
+	)
+	if err != nil {
+		t.Fatal("Expected no error while encrypting stream with key ring, got:", err)
+	}
+	reachedEnd := false
+	bufferSize := 2
+	buffer := make([]byte, bufferSize)
+	for !reachedEnd {
+		n, err := messageReader.Read(buffer)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				reachedEnd = true
+			} else {
+				t.Fatal("Expected no error while reading data, got:", err)
+			}
+		}
+		writtenTotal := 0
+		for writtenTotal < n {
+			written, err := messageWriter.Write(buffer[writtenTotal:n])
+			if err != nil {
+				t.Fatal("Expected no error while writing data, got:", err)
+			}
+			writtenTotal += written
+		}
+	}
+	err = messageWriter.Close()
+	if err != nil {
+		t.Fatal("Expected no error while closing plaintext writer, got:", err)
+	}
+	decryptedReader, err := keyRingPrivate.DecryptStream(
+		&ciphertextBuf,
+		keyRingPublic,
+		GetUnixTime(),
+	)
+	if err != nil {
+		t.Fatal("Expected no error while calling decrypting stream with key ring, got:", err)
+	}
+	decryptedBytes, err := io.ReadAll(decryptedReader)
+	if err != nil {
+		t.Fatal("Expected no error while reading the decrypted data, got:", err)
+	}
+	if !bytes.Equal(decryptedBytes, messageBytes) {
+		t.Fatalf("Expected the decrypted data to be %s got %s", string(decryptedBytes), string(messageBytes))
+	}
+	err = decryptedReader.VerifySignature()
+	if err != nil {
+		t.Fatal("Expected no error while verifying the signature, got:", err)
+	}
+	if isBinary != decryptedReader.IsBinary() {
+		t.Fatalf("Expected isBinary to be %t got %t", isBinary, decryptedReader.IsBinary())
+	}
+	if filename != decryptedReader.GetFilename() {
+		t.Fatalf("Expected filename to be %s got %s", filename, decryptedReader.GetFilename())
+	}
+	if modTime != decryptedReader.GetModificationTime() {
+		t.Fatalf("Expected modification time to be %d got %d", modTime, decryptedReader.GetModificationTime())
+	}
+}
+
+func TestKeyRing_EncryptSplitStream(t *testing.T) {
```

Comment:
I'd also test reading the encrypted data packet and key packet, join them then try decrypting with the "normal" functions

Reply:
Done